### PR TITLE
Upgrade Kafka+Zookeeper Docker Containers

### PR DESCRIPTION
**Pull request by gherceg**
## Technical Summary

https://dimagi-dev.atlassian.net/browse/SAAS-13370

- targeting Kafka v3.2
- targeting Zookeeper v3.7 with the hope to remove the need for this container entirely in the near future in favor of kraft

I was concerned about [compatibility with kafka-python](https://kafka-python.readthedocs.io/en/master/compatibility.html), but it sounds like kafka has backwards compatibility:
> Because the kafka server protocol is backwards compatible, kafka-python is expected to work with newer broker releases as well.
>
> Although kafka-python is tested and expected to work on recent broker versions, not all features are supported. Specifically, authentication codecs, and transactional producer/consumer support are not fully implemented. PRs welcome!

`corehq.form_processor.tests.test_kafka` and `corehq.apps.change_feed.tests.test_kafka_change_feed` exercise the kafka broker, and still pass after these changes.

We definitely need to run through QA on staging when the Server Infrastructure pod rolls this out.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: docker/hq-compose.yml [hunk 1]
```diff
@@ -110,14 +110,14 @@ services:
...
+    image: zookeeper:3.7
     ports:
       - "2181:2181"
     volumes:
-      - ${VOLUME_PREFIX}${BLANK_IF_TESTS-zookeeper:}/opt/zookeeper-3.4.13/data
+      - ${VOLUME_PREFIX}${BLANK_IF_TESTS-zookeeper:}/opt/zookeeper-3.7/data
 
   kafka:
-      image: wurstmeister/kafka:2.13-2.6.0
+      image: bitnami/kafka:3.2
```
Comment (joxl) [inline docker/hq-compose.yml:120]: I assume this is a trusted/vetted container source?

**Review by joxl — COMMENTED** [review]
If you come up with any novel workflows (or discover existing ones) to verify that these containers are "doing everything they need to do", I'm very interested.

File: docker/hq-compose.yml (hunk 1 above)
Comment (gherceg) [reply docker/hq-compose.yml:120]: Yeah they are a verified publisher with 100+ million downloads of that specific image, as well as lots of downloads of other images. Bitnami is owned by VMWare fwiw.

**Review by millerdev — APPROVED** [review]

**Comment by gherceg** [comment]
We should be good to move forward with the steps outlined in https://confluence.dimagi.com/display/saas/Upgrading+CommCare+HQ%27s+backing+services.

@dimagi/dimagi-dev if you'd like to start running updated versions of kafka and zookeeper locally, follow the steps outlined [here](https://gist.github.com/gherceg/1f0b7645cce6f1f8c53b9e57714d495e).

**Comment by snopoke** [comment]
> We should be good to move forward with the steps outlined in https://confluence.dimagi.com/display/saas/Upgrading+CommCare+HQ%27s+backing+services.
> 
> if you'd like to start running updated versions of kafka and zookeeper locally, follow the steps outlined [here](https://gist.github.com/gherceg/1f0b7645cce6f1f8c53b9e57714d495e).

Can you re-post on Slack once this is merged?

**Comment by gherceg** [comment]
> Can you re-post on Slack once this is merged?

Yep will do.

**Comment by gherceg** [comment]
SaaS has completed [the upgrade](https://dimagi-dev.atlassian.net/browse/SAAS-13630), so I am merging this.

@dimagi/dimagi-dev once on the latest version of master, run the following:
```
./scripts/docker stop kafka zookeeper
./scripts/docker up -d kafka zookeeper
```
To confirm the upgrade was successful, you can run `docker ps` and ensure `bitnami/kafka:3.2` and `zookeeper:3.7` containers are running rather than `wurstmeister/kafka:2.13-2.6.0` and `wurstmeister/zookeeper`.

**Pull request merged**